app/controllers/puppies_controller.rb

```diff
@@ -4,7 +4,7 @@ class PuppiesController < ApplicationController
 
   def index
     if !params[:city].nil? && !params[:city].empty?
-      @puppies = Puppy.where("city = ?", params[:city])
+      @puppies = Puppy.where("city LIKE ?", "%#{params[:city].capitalize}%")
     else
       @puppies = Puppy.all
     end
```

Comment:
Can we please show a result message, especially when there are "no puppies" for a search.
Otherwise it looks like an error.